receipt.go

```diff
@@ -1,15 +1,65 @@
 package ofdconnector
 
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/billz-2/ofd_connector/internal/constants"
+	"github.com/billz-2/ofd_connector/internal/gateway"
+)
+
+const (
+	receiptInfoEndpoint         = "/FiscalDrive/Receipt/Info/"
+	receiptRegisterTXIDEndpoint = "/FiscalDrive/Receipt/RegisterTXID/"
+	receiptGetTXIDEndpoint      = "/FiscalDrive/Receipt/GetTXID/"
+
+	databaseFilesCountEndpoint = "/Database/Files/Count"
+	databaseFilesStatusReset   = "/DataBase/Files/Status/Reset"
+)
+
+type ReceiptI interface {
+	GetTXID(ctx context.Context, req SaleParams) (int64, error)
+	RegisterTXID(ctx context.Context, txID int64) (ReceiptInfo, error)
+	GetReceiptInfo(ctx context.Context, index uint32) (ReceiptFullInfo, error)
+	GetDatabaseFilesCount(ctx context.Context, status uint16) (map[string]int64, error)
+	ResetDatabaseFilesStatus(ctx context.Context, txID int64) error
+}
+
+type receiptConfig struct {
+	Gateway gateway.Client
```

Comment:
This is minor but i clearly don't understand do we need those field exportable or not? cause in ZReport case gateway is `unexportable` here it is `exportable`. what's the difference?

Reply:
probably, that time I had confusion that this will be called from outside of the package. Made unexportable now